schemahcl/extension.go

```diff
@@ -547,10 +547,16 @@ func scanAttr(key string, r *Resource, field reflect.Value) error {
 	return nil
 }
 
+// specCache holds a cache for type-spec fields descriptions.
+var specCache sync.Map
+
```

Comment:
Is this called concurrently? A simple map is faster.

Reply:
Yes - our tests :D